packages/flame_3d/lib/src/model/model_animation.dart

```diff
@@ -0,0 +1,145 @@
+import 'dart:math';
+
+import 'package:flame_3d/core.dart';
+import 'package:flame_3d/src/parser/gltf/animation_interpolation.dart';
+
+abstract class AnimationSpline<T> {
+  final AnimationInterpolation interpolation;
+  final List<(double, T)> values;
```

Comment:
I would have this as a tuple with named arguments so that it is easier to read further down, probably typedef it?